launcher/Version.cpp

```diff
@@ -74,12 +74,36 @@ bool Version::operator!=(const Version &other) const
 void Version::parse()
 {
     m_sections.clear();
-
-    // FIXME: this is bad. versions can contain a lot more separators...
-    QStringList parts = m_string.split('.');
-
-    for (const auto& part : parts)
-    {
-        m_sections.append(Section(part));
+    QString currentSection;
+    bool lastCharWasDigit = false;
+    for (int i = 0; i < m_string.size(); ++i) {
+        if(m_string[i].isDigit()){
+            if(!lastCharWasDigit){
```

Comment:
Hmm I think the logic could be cleaned up significantly but we're getting there...

```c++

auto class_change = [] (QChar lastChar, QChar currentChar) {
    return (( lastChar.isLetter() && currentChar.isDigit() ) || (lastChar.isDigit() && currentChar.isLetter()) );
} 
```

Using an anonymous function like this would prevent the code duplication and imo make the logic clearer.

Reply:
done in https://github.com/PrismLauncher/PrismLauncher/pull/751/commits/f2ad6c5eef3e54206257dfd92369b8e49954fd79

Reply:
done in https://github.com/PrismLauncher/PrismLauncher/pull/751/commits/b58a9b32f65558ff61ed91259b309c2ece2ddab8